### ISSUE TITLE
passing callback to `observeSteps` causes crash

### ISSUE THREAD

**Issue by 256hz** [StasDoskalenko/react-native-google-fit#328]
Hi there,

I would like to periodically (every ~15 min) get step count while the app is backgrounded.  Is that possible with `observeSteps`?

I was hoping to test this, but the following line causes the app to crash:

`GoogleFit.observeSteps((isError, result) => console.log({ isError, result });`

When I look at the code for `observeSteps`, it looks like there are no arguments in the method signature:

```java
    @ReactMethod
    public void observeSteps() {
        mGoogleFitManager.getStepCounter().findFitnessDataSources();
    }
```

Is this feature intended for use in the way I'm hoping?

RN: 69.1
react-native-google-fit: 0.18.3 (but latest version has the same code)

**Comment by aboveyunhai** [comment]
@256hz  yes for 2 years ago, because the permission request from javascript side was written by me at that moment.
But a lot of codes might be obsolete and require full rewrite after Google made a ton of changes since android 10

**Comment by aboveyunhai** [comment]
by the way, my fix was pretty immature at that time.

**Comment by aboveyunhai** [comment]
@256hz  it's fine, I would leave it open in case ppl encounter the similar issues. And the problem is not solved yet regardless.
